### PR TITLE
iptables: Added set module/match_set support

### PR DESCRIPTION
**Pull request by mzima**
##### SUMMARY
Added support for the iptables "set" extension to allow the use of ipsets 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- lib/ansible/modules/iptables.py

### CODE REVIEW

**Comment by larrythecow** [comment]
+1

**Comment by mzima** [comment]
Would it be possible to merge this PR? Otherwise I've to cope with the stale_ci label every other week or so ;) Let me know if there are any other things I should to ...

**Comment by samdoran** [comment]
@mzima You don't have to worry about `stale_ci`. If tests passed, we'll rerun them before merging.

This PR sparked some internal discussion around the need to have a good way to allow adding support for `iptables` extensions in a way that would fail nicely if the extension was not installed. I [opened a feature request](https://github.com/ansible/ansible/issues/73012) for that.

I was trying to determine if the `set` extension is in the default extensions provided by `iptables`. It seems like it is, but I wasn't able to 100% verify that. Do you know if this is in the default set of extensions?

**Comment by mzima** [comment]
>**Do you know if this is in the default set of extensions?**
Yes it is. Since 1.2.10 (http://www.netfilter.org/projects/iptables/files/changes-iptables-1.2.10.txt) which was released back in 2004.

File: lib/ansible/modules/iptables.py [hunk 1]
```diff
@@ -290,6 +290,22 @@
       - Specifies the destination IP range to match in the iprange module.
     type: str
     version_added: "2.8"
+  match_set:
+    description:
+      - Specifies a set name which can be defined by ipset.
+      - Must be used together with the match_set_flags parameter.
+      - When the C(!) argument is prepended then it inverts the rule.
+      - Uses the iptables set module.
```
Comment (samdoran) [inline lib/ansible/modules/iptables.py]:
```suggestion
      - Uses the iptables set extension.
```

File: lib/ansible/modules/iptables.py [hunk 2]
```diff
@@ -290,6 +290,22 @@
...
+    type: str
+    version_added: "2.11"
+  match_set_flags:
+    description:
+      - Specifies the necessary flags for the match_set parameter.
+      - Must be used together with the match_set parameter.
+      - Uses the iptables set module.
```
Comment (samdoran) [inline lib/ansible/modules/iptables.py]:
```suggestion
      - Uses the iptables set extension.
```

File: test/units/modules/test_iptables.py [hunk 3]
```diff
@@ -953,3 +953,38 @@ def test_destination_ports(self):
...
             '--comment', 'this is a comment'
         ])
+
+    def test_match_set(self):
+        """ Test match_set together with match_set_flags """
+        set_module_args({
+            'chain': 'INPUT',
+            'protocol': 'tcp',
+            'match_set': 'admin_hosts',
+            'match_set_flags': 'src',
```
Comment (samdoran) [inline test/units/modules/test_iptables.py:963]: Do we need additional test cases for `dst`, `dst,src`, and `src,dst`?

File: lib/ansible/modules/iptables.py (hunk 1 above)
Comment (mzima) [reply lib/ansible/modules/iptables.py]: Ok -> Changed in  18a72bc.

File: lib/ansible/modules/iptables.py (hunk 2 above)
Comment (mzima) [reply lib/ansible/modules/iptables.py]: Ok -> Changed in  18a72bc.

File: test/units/modules/test_iptables.py (hunk 3 above)
Comment (mzima) [reply test/units/modules/test_iptables.py:963]: Added an additional `src,dst` test in 18a72bc. More test are probably overkill. Let me know if this is OK for you ...

**Review by samdoran — APPROVED** [review]

**Pull request merged**